From 8c386a9a6bedc1da910c965f0eee76f65c1ba1dd Mon Sep 17 00:00:00 2001
From: Han Xu <han.xu@nxp.com>
Date: Thu, 1 Nov 2018 15:01:27 -0500
Subject: [PATCH 4973/5242] MLK-20171: mtd: flexspi: reset flexspi FLASHxCR2
 registers during probe

commit  a94e8154cd7824b8a01638f45403c38a2dce0b1b from
https://source.codeaurora.org/external/imx/linux-imx.git

Flexspi registers cannot be reset to default value, reset all FLASHxCR2
registers to 0 to avoid read data with invalid LUT commands.

Signed-off-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mtd/spi-nor/fsl-flexspi.c |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/mtd/spi-nor/fsl-flexspi.c b/drivers/mtd/spi-nor/fsl-flexspi.c
index c7b4eae..9519749e 100644
--- a/drivers/mtd/spi-nor/fsl-flexspi.c
+++ b/drivers/mtd/spi-nor/fsl-flexspi.c
@@ -944,7 +944,6 @@ static int fsl_flexspi_init_rpm(struct fsl_flexspi *flex)
 static int fsl_flexspi_nor_setup(struct fsl_flexspi *flex)
 {
 	void __iomem *base = flex->iobase;
-	u32 reg;
 
 	/* Reset the module */
 	writel(FLEXSPI_MCR0_SWRST_MASK, base + FLEXSPI_MCR0);
@@ -959,8 +958,11 @@ static int fsl_flexspi_nor_setup(struct fsl_flexspi *flex)
 	writel(FLEXSPI_MCR0_AHB_TIMEOUT_MASK | FLEXSPI_MCR0_IP_TIMEOUT_MASK |
 	       FLEXSPI_MCR0_OCTCOMB_EN_MASK, base + FLEXSPI_MCR0);
 
-	/* Read the register value */
-	reg = readl(base + FLEXSPI_MCR0);
+	/* Reset the FLASHxCR2 */
+	writel(0, base + FLEXSPI_FLSHA1CR2);
+	writel(0, base + FLEXSPI_FLSHA2CR2);
+	writel(0, base + FLEXSPI_FLSHB1CR2);
+	writel(0, base + FLEXSPI_FLSHB2CR2);
 
 	/* Init the LUT table. */
 	fsl_flexspi_init_lut(flex);
-- 
1.7.9.5

